src/vulkan-renderer/world/collision.cpp

```diff
@@ -0,0 +1,184 @@
+                                       cube_face{"back", adjust_coordinates(cube_directions[3])},
+                                       cube_face{"top", adjust_coordinates(cube_directions[4])},
+                                       cube_face{"bottom", adjust_coordinates(cube_directions[5])}};
+
+    // The coordinates of every corner of the cube.
+    std::array cube_corners = {cube_corner{"left back top", adjust_coordinates({-1.0f, 1.0f, 1.0f})},
+                               cube_corner{"right back top", adjust_coordinates({1.0f, 1.0f, 1.0f})},
+                               cube_corner{"left back botom", adjust_coordinates({-1.0f, 1.0f, -1.0f})},
+                               cube_corner{"right back bottom", adjust_coordinates({1.0f, 1.0f, -1.0f})},
+                               cube_corner{"left front top", adjust_coordinates({-1.0f, -1.0f, 1.0f})},
+                               cube_corner{"right front top", adjust_coordinates({1.0f, -1.0f, 1.0f})},
+                               cube_corner{"left front bottom", adjust_coordinates({-1.0f, -1.0f, -1.0f})},
+                               cube_corner{"right front bottom", adjust_coordinates({1.0f, -1.0f, -1.0f})}};
+
+    using corner_on_face_index = std::array<std::size_t, 4>;
+
+    // These indices specify which 4 points describe the corners on the given face.
+    static constexpr std::array CORNERS_ON_FACE_INDICES{
+        corner_on_face_index{0, 2, 4, 6}, // left
+        corner_on_face_index{1, 3, 5, 7}, // right
+        corner_on_face_index{4, 5, 6, 6}, // front
+        corner_on_face_index{0, 1, 2, 3}, // back
+        corner_on_face_index{0, 1, 4, 5}, // top
+        corner_on_face_index{2, 3, 6, 7}  // bottom
+    };
+
+    // The coordinates of the center of every edge of the cube.
+    std::array cube_edges = {cube_edge{"left top", adjust_coordinates({-1.0f, 0.0f, 1.0f})},
+                             cube_edge{"left front", adjust_coordinates({-1.0f, 1.0f, 0.0f})},
+                             cube_edge{"left bottom", adjust_coordinates({-1.0f, 0.0f, -1.0f})},
+                             cube_edge{"left back", adjust_coordinates({-1.0f, -1.0f, 0.0f})},
+                             cube_edge{"right top", adjust_coordinates({1.0f, 0.0f, 1.0f})},
+                             cube_edge{"right front", adjust_coordinates({1.0f, 1.0f, 0.0f})},
+                             cube_edge{"right bottom", adjust_coordinates({1.0f, 0.0f, -1.0f})},
+                             cube_edge{"right back", adjust_coordinates({1.0f, -1.0f, 0.0f})},
+                             cube_edge{"middle bottom front", adjust_coordinates({0.0f, -1.0f, -1.0f})},
+                             cube_edge{"middle bottom back", adjust_coordinates({0.0f, 1.0f, -1.0f})},
+                             cube_edge{"middle top front", adjust_coordinates({0.0f, -1.0f, 1.0f})},
+                             cube_edge{"middle top back", adjust_coordinates({0.0f, 1.0f, 1.0f})}};
```

Comment:
Same here, according to our documentation the middle would be `0.5`.

Reply:
I clarified in the commit that this is not the cube data, but the bounding boxes data around the cube. It is very practical to have `(0,0,0)` as center of the bounding box.

Reply:
Done.